### PR TITLE
Call extension code to import browser bookmarks for importing to library

### PR DESCRIPTION
**Pull request by yipingliao**
@andrewconner 

Thank you @2is10 for the extension support and explanation.

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js [hunk 1]
```diff
@@ -158,17 +158,35 @@ angular.module('kifi')
...
+        if (!kifiVersion) {
           modalService.open({
             template: 'common/modal/importBookmarksErrorModal.tpl.html'
           });
+          return;
+        }
+
+        $analytics.eventTrack('user_clicked_page', {
+          'type': 'browserImport',
+          'action': 'ImportToLibrary'  // TODO: update this when we have the full tracking spec.
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: We need to be careful about sending event data to mixpanel that hasn’t been finalized/approved by product (Jen). Any strings we use even once will forever appear in dropdowns and (sometimes) in charts, which can be really annoying for people who use our mixpanel account a lot.

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js [hunk 2]
```diff
@@ -158,17 +158,35 @@ angular.module('kifi')
...
+        });
+
+        var event = $scope.msgEvent && $scope.msgEvent.origin && $scope.msgEvent.source && $scope.msgEvent;
+        var message = {
+          type: 'import_bookmarks',
+          libraryId: library.id
+        };
+
+        if (event) {
+          event.source.postMessage(message, $scope.msgEvent.origin);
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: `$scope.msgEvent` → `event`

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js [hunk 3]
```diff
@@ -158,17 +158,35 @@ angular.module('kifi')
...
           });
+          return;
+        }
+
+        $analytics.eventTrack('user_clicked_page', {
+          'type': 'browserImport',
+          'action': 'ImportToLibrary'  // TODO: update this when we have the full tracking spec.
+        });
+
+        var event = $scope.msgEvent && $scope.msgEvent.origin && $scope.msgEvent.source && $scope.msgEvent;
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: The sanity checking doesn’t appear to be necessary. `$scope.msgEvent` should be falsy if not a valid message event.

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js [hunk 4]
```diff
@@ -158,17 +158,35 @@ angular.module('kifi')
...
       // the next modal.
       $scope.$evalAsync(function () {
-        // Fake check. To do: use real endpoints to determine which modal to open.
-        if (library) {
-          modalService.open({
-            template: 'common/modal/importBookmarksLibraryInProgressModal.tpl.html',
-            scope: $scope
-          });
-        } else {
+        var kifiVersion = $window.document.getElementsByTagName('html')[0].getAttribute('data-kifi-ext');
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: `.documentElement` is short for `.getElementsByTagName('html')[0]`

**Comment by 2is10** [comment]
noted a few minor style suggestions... looks good!

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js (hunk 1 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: Ugh, that's true. I'll comment out this whole section.

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js (hunk 4 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: Will update. Thanks!

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js (hunk 2 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: Will update. Thanks!

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js (hunk 3 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: Just to make sure: a simple `var event = $scope.msgEvent;` should do?

Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: Yes.​

**Pull request merged**